opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -152,23 +175,29 @@ impl<T: Number<T>> Sum<T> {
             });
         }
 
+        let mut values = match self.value_map.values.write() {
+            Ok(v) => v,
+            Err(_) => return (0, None),
+        };
+
+        let mut seen = HashSet::new();
         for (attrs, value) in values.drain() {
-            s_data.data_points.push(DataPoint {
-                attributes: attrs
-                    .iter()
-                    .map(|(k, v)| KeyValue::new(k.clone(), v.clone()))
-                    .collect(),
-                start_time: Some(prev_start),
-                time: Some(t),
-                value: value.get_value(),
-                exemplars: vec![],
-            });
+            if seen.insert(Arc::as_ptr(&value)) {
```

Comment:
Dedup sorted+original attributes!
@utpilla I leveraged Arc pointer to de-dup. Storing MP in diff. array was tried as well, but this looks good to me.